spec/spec_helper.rb

```diff
@@ -52,3 +52,80 @@ def expect_push_bulk_to_match(klass, matcher)
     'args' => matcher,
   }))
 end
+
+class StreamingServerManager
+  @running_thread = nil
+
+  def initialize
+    at_exit { stop }
+  end
+
+  def start(port: 4020)
+    return if @running_thread
+
+    queue = Queue.new
+
+    @queue = queue
+
+    @running_thread = Thread.new do
+      Open3.popen2e(
+        {
+          'REDIS_NAMESPACE' => ENV.fetch('REDIS_NAMESPACE'),
+          'DB_NAME' => "#{ENV.fetch('DB_NAME', 'mastodon')}_test#{ENV.fetch('TEST_ENV_NUMBER', '')}",
+          'RAILS_ENV' => ENV.fetch('RAILS_ENV', 'test'),
+          'NODE_ENV' => ENV.fetch('NODE_ENV', 'test'),
+          'PORT' => port.to_s,
+        },
+        'node index.js', # must not call yarn here, otherwise it will fail because yarn does not send signals to its child process
```

Comment:
Is the main reason this class needs to exist so that the rspec run can manage the start/stop of the streaming server appropriately?

I'm assuming it would be straightforward on CI to manage it more directly, but that needing it to start/stop within the spec suite is why this approach was chosen?

Reply:
This is not only for CI, but you need to run the streaming server, against the correct DB, when running the test suite locally as well. I found it to be easier to automatically handle this rather than writing docs & checks to ensure it has been started correctly by the user